### PR TITLE
Bug fix/make graphs transaction aware

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Make actions in /_api/gharial transaction aware, so they can run inside cluster transactions as well.
Added tests for el cheapo and JS transactions.

Enterprise companion PR: https://github.com/arangodb/enterprise/pull/301

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [x] Added new **integration tests** (shell_client tests/js/client/shell/shell-graph-transaction.js)

https://172.16.10.101/view/PR/job/arangodb-matrix-pr/5923/

### CODE REVIEW

File: arangod/Graph/GraphOperations.cpp [hunk 1]
```diff
@@ -902,7 +904,7 @@ OperationResult GraphOperations::removeEdgeOrVertex(const std::string& collectio
 
       arangodb::aql::Query query(false, _vocbase, queryString, bindVars,
                                  nullptr, arangodb::aql::PART_DEPENDENT);
-      query.setTransactionContext(ctx);  // hack to share  the same transaction
+      query.setTransactionContext(_ctx);  // hack to share the same transaction
```
Comment (graetzer) [inline arangod/Graph/GraphOperations.cpp]:
```suggestion
      query.setTransactionContext(ctx());  // hack to share the same transaction
```

**Comment by jsteemann** [comment]
We probably need to add additional tests for smart graphs here as well.

**Comment by jsteemann** [comment]
https://172.16.10.101/view/PR/job/arangodb-matrix-pr/5951/

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5956/

**Pull request merged**